src/lib/components/Subscribe/index.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * @fileoverview Element that renders newsletter subscription form.
+ */
+
+import {BaseElement} from "../BaseElement";
+import "./_styles.scss";
+
+/**
+ * Element that renders newsletter subscription form.
+ *
+ * @extends {BaseElement}
+ * @final
+ */
+class Subscribe extends BaseElement {
+  constructor() {
+    super();
+    this.checkboxes = ["WebDevNewsletter", "collects-pii-spii-checkbox"];
+    this.needsDoubleOptIn = [
+      "AT: Austria",
+      "DE: Germany",
+      "GR: Greece",
+      "LU: Luxembourg",
+      "NO: Norway",
+    ];
+    this.robotName = "is-it-just-me-or-was-this-form-filled-out-by-a-robot";
+    this.processing = false;
+    this.submitted = false;
+    this.onSubmit = this.onSubmit.bind(this);
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.form = this.querySelector(".w-subscribe__form");
+    this.subscribeError = this.querySelector(".w-subscribe__error");
+    this.subscribeMessage = this.querySelector(".w-subscribe__message");
+    this.submissionUrl = this.form.action;
+    this.form.addEventListener("submit", this.onSubmit);
+  }
+
+  detachedCallback() {
+    super.detachedCallback();
+    this.form.removeEventListener("submit", this.onSubmit);
+  }
+
+  /**
+   * @param {FormData} form
+   * @return {FormData}
+   */
+  cleanForm(form) {
+    const doubleOptIn = this.needsDoubleOptIn.includes(form.get("Country"));
+    this.checkboxes.forEach((checkbox) =>
+      form.set(checkbox, doubleOptIn ? "Unconfirmed" : "True"),
+    );
+    form.delete(this.robotName);
+    return form;
+  }
+
+  postForm(body) {
+    return fetch(this.submissionUrl, {
+      method: "POST",
+      body,
+    }).then((r) => r.json());
+  }
+
+  onError(errors) {
+    this.subscribeError.textContent = "";
+    if (errors) {
+      Object.values(errors).forEach((e) => {
+        const pTag = document.createElement("p");
+        pTag.innerText = typeof e === "string" ? e : e.join(" ");
+        this.subscribeError.appendChild(pTag);
+      });
+    }
+  }
+
+  onSubmit(e) {
+    e.preventDefault();
+    if (this.processing || this.submitted) {
+      return false;
+    }
+    this.processing = true;
+    const form = new FormData(e.target);
+    const formIsRobot = form.get(this.robotName).length !== 0;
+
+    if (formIsRobot) {
+      this.onSuccess();
+      return false;
+    }
+    const cleanedForm = this.cleanForm(form);
+
+    this.postForm(cleanedForm)
+      .then((response) => {
+        if (response && response.result === "accepted") {
+          this.onSuccess();
```

Comment:
To be safe I would suggest binding `this.onSuccess` and `this.onError` in the constructor, similar to what you did for `this.onSubmit`. Technically you don't need it because the value of `this` inside of the `onSubmit` function is pointing at the instance that has `onSuccess` and `onError` defined on it, but if that ever changed then it might appear to break mysteriously.